packages/discount-code-importer/src/main.js

```diff
@@ -0,0 +1,245 @@
+              this.logger.error(msg)
+              return Promise.resolve()
+            }
+            // eslint-disable-next-line max-len
+            const msg = 'Process stopped due to error while creating discount code. See summary for details'
+            this.logger.error(msg)
+            this._summary.updateErrorCount += 1
+            this._summary.errors.push(
+              error.message ? error.message : error,
+            )
+            return Promise.reject(error)
+          })
+      return this._create(newCode)
+        .then(() => {
+          this._summary.created += 1
+          return Promise.resolve()
+        })
+        .catch((error) => {
+          if (this.continueOnProblems) {
+            this._summary.createErrorCount += 1
+            this._summary.errors.push(
+              error.message ? error.message : error,
+            )
+            // eslint-disable-next-line max-len
+            const msg = 'Create error occured but ignored. See summary for details'
+            this.logger.error(msg)
+            return Promise.resolve()
+          }
+          // eslint-disable-next-line max-len
+          const msg = 'Process stopped due to error while creating discount code. See summary for details'
+          this.logger.error(msg)
+          this._summary.createErrorCount += 1
+          this._summary.errors.push(
+            error.message ? error.message : error,
+          )
+          return Promise.reject(error)
+        })
+    })
+  }
+
+  _update (newCode: CodeData, existingCode: CodeData) {
+    const actions = this.syncDiscountCodes.buildActions(newCode, existingCode)
+    // don't call API if there's no update action
+    if (!actions.length) return Promise.resolve({ statusCode: 304 })
+    const service = this._createService()
+    const uri = service.byId(existingCode.id).build()
+    const req = {
+      uri,
+      method: 'POST',
+      body: { version: existingCode.version, actions },
+    }
+    this.logger.verbose('Updating existing code entry')
+    return this.client.execute(req)
+  }
+
+  _create (code: CodeData) {
+    const service = this._createService()
+    const uri = service.build()
+    const req = {
+      uri,
+      method: 'POST',
+      body: code,
+    }
+    this.logger.verbose('Creating new code entry')
+    return this.client.execute(req)
+  }
+
+  _createService () {
+    return createRequestBuilder({
+      projectKey: this.apiConfig.projectKey,
+    }).discountCodes
+  }
+
+  _resetSummary () {
+    this._summary = {
+      created: 0,
+      updated: 0,
+      unchanged: 0,
+      createErrorCount: 0,
+      updateErrorCount: 0,
+      errors: [],
+    }
+    return this._summary
+  }
+
+  summaryReport () {
+    const {
+      created,
+      updated,
+      unchanged,
+      createErrorCount,
+      updateErrorCount,
+    } = this._summary
+    let message = ''
+    if (created === 0 && updated === 0 &&
+      createErrorCount === 0 && updateErrorCount === 0)
```

Comment:
What about this shortcut?
`created + updated + createErrorCount + updateErrorCount === 0`